src/netml/pparser/parser.py

```diff
@@ -971,24 +987,18 @@ def _iter_pcap_dict(self):
         """
         with PcapReader(self.pcap_file) as pcap_reader:
             for pkt in pcap_reader:
-                if Ether not in pkt:
-                    continue
-
-                pkt_dict = {
-                    'time': pkt.time,
-                    'datetime': datetime.fromtimestamp(int(pkt.time)),
-                    'length': len(pkt),
-                    'mac_dst': pkt[Ether].dst,
-                    'mac_src': pkt[Ether].src,
-                    'ip_dst': None,
-                    'ip_src': None,
-                    'protocol': None,
-                    'port_dst': None,
-                    'port_src': None,
-                    'is_dns': False,
-                    'dns_query': None,
-                    'dns_resp': None,
-                }
+                pkt_dict = dict.fromkeys(self._pcap_columns)
+
+                pkt_dict.update(
+                    time=pkt.time,
+                    datetime=datetime.fromtimestamp(int(pkt.time)),
+                    length=len(pkt),
+                    is_dns=False,
+                )
+
+                if Ether in pkt:
+                    pkt_dict['mac_dst'] = pkt[Ether].dst
+                    pkt_dict['mac_src'] = pkt[Ether].src
```

Comment:
Presumably, it's not unreasonable to retain packets which are missing MAC address information; (indeed, this was the issue at the heart of #31).

However, right or wrong, this method was skipping such packets entirely. As such, these elements were discarded; (and, in that case, the resulting stream was entirely _empty_).

Instead, this proposes to retain such packets, and simply allow these columns, like others, to be `None`, unless these data are provided.